handler/src/test/java/io/netty/handler/ssl/PemEncodedTest.java

```diff
@@ -69,6 +72,26 @@ private static void testPemEncoded(SslProvider provider) throws Exception {
         }
     }
 
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
fwiw, I would avoid using `expected`, because it isn't clear what the error is.  Below, getAlgorithm returns null, which might also trigger IllegalArgumentException.  I can't really tell without digging into the source.  

Instead, using `ExpectedException` JUnit rule let's you specify both the exception type, and match a partial string of the error message.  It's more resilient to catching the wrong exception, and it's easier to read.

Reply:
Time to migrate to a better option like AssertJ or Spock?

Reply:
```java
assertThatThrownBy(() -> {
    ...
}).isInstanceOf(IllegalArgumentException.class);
```

Reply:
I haven't used either, though my preferred option `Truth` says it's similar to AssertJ.  JUnit5 has `assertThrows` which could also be used.

Reply:
I'm a big fan of AssertJ due to its method chaining design which reduces the amount of imports. Alternatively, Spock may be a good choice although it's Groovy.

Reply:
Yeah, it looks like even the [Truth](https://google.github.io/truth/comparison) page likes AssertJ better.

Reply:
Honestly I dont care too much... its "just" a testing library. So whatever fits us is ok by me. I think the important is to be just consistent and us either A or B and not end up using everything.

Reply:
@carl-mastrangelo please feel free to submit a PR using `@expected` here.